OpenRA.Mods.Common/Traits/Crates/Crate.cs

```diff
@@ -76,10 +73,14 @@ public void OnCrush(Actor crusher)
 				var n = self.World.SharedRandom.Next(totalShares);
 
 				foreach (var s in shares)
-				{
+				{
 					if (n < s.Second)
 					{
+						if (!crusher.HasTrait<Armament>() && s.First.GetEffectName().Equals("firepower"))
```

Comment:
This doesn't appear to be related to the hotkey changes, and has its own issues that are best dealt with in its own pr.  Could you please split it off from this one?